rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java

```diff
@@ -598,6 +631,38 @@ private Stream<Arguments> expectedCreationBlockNumberArgs() {
                 Arguments.of(blockNumberFederationActivationFingerroot, hopActivations, newFederation.getCreationBlockNumber())
             );
         }
+
+        @ParameterizedTest
+        @Tag("getActiveFederatorBtcPublicKey")
+        @MethodSource("expectedFederatorBtcPublicKeyArgs")
+        void getActiveFederatorBtcPublicKey_returnsExpectedFederatorBtcPublicKeyAccordingToActivationAgeAndActivations(long currentBlock, ActivationConfig.ForBlock activations, byte[] expectedFederatorBtcPublicKey) {
+            Block executionBlock = mock(Block.class);
+            when(executionBlock.getNumber()).thenReturn(currentBlock);
+
+            federationSupport = federationSupportBuilder
+                .withFederationConstants(federationMainnetConstants)
+                .withFederationStorageProvider(storageProvider)
+                .withRskExecutionBlock(executionBlock)
+                .withActivations(activations)
+                .build();
+
+            byte[] activeFederatorBtcPublicKey = federationSupport.getActiveFederatorBtcPublicKey(0);
+            assertThat(activeFederatorBtcPublicKey, is(expectedFederatorBtcPublicKey));
+        }
+
+        private Stream<Arguments> expectedFederatorBtcPublicKeyArgs() {
+            BtcECKey federatorFromOldFederationBtcPublicKey = oldFederation.getBtcPublicKeys().get(0);
+            BtcECKey federatorFromNewFederationBtcPublicKey = newFederation.getBtcPublicKeys().get(0);
+
+            return Stream.of(
+                Arguments.of(blockNumberFederationActivationHop - 1, hopActivations, federatorFromOldFederationBtcPublicKey.getPubKey()),
+                Arguments.of(blockNumberFederationActivationHop, hopActivations, federatorFromNewFederationBtcPublicKey.getPubKey()),
+                Arguments.of(blockNumberFederationActivationHop, fingerrootActivations, federatorFromOldFederationBtcPublicKey.getPubKey()),
+                Arguments.of(blockNumberFederationActivationFingerroot - 1, fingerrootActivations, federatorFromOldFederationBtcPublicKey.getPubKey()),
+                Arguments.of(blockNumberFederationActivationFingerroot, fingerrootActivations, federatorFromNewFederationBtcPublicKey.getPubKey()),
+                Arguments.of(blockNumberFederationActivationFingerroot, hopActivations, federatorFromNewFederationBtcPublicKey.getPubKey())
```

Comment:
I would rather pass the `BtcECKey` instead of the `byte[]`, since a `byte[]` can be anything, but a `BtcECKey` can only be a `BtcECKey`.
Then, inside of the test, you do `btcEcKey.getPubKey()`.

Reply:
yep, good catch. Thanks jere!